src/propertiesdialog.cpp

```diff
@@ -188,7 +190,17 @@ PropertiesDialog::PropertiesDialog(QWidget *parent)
     trimPastedTrailingNewlinesCheckBox->setChecked(Properties::Instance()->trimPastedTrailingNewlines);
     confirmMultilinePasteCheckBox->setChecked(Properties::Instance()->confirmMultilinePaste);
 
-    resize(sizeHint()); // show it compact but not too much
+    // show it compact inside available desktop geometry
+    QSize ag;
+    if (parent != nullptr)
+    {
+        if (QWindow *win = parent->windowHandle())
+        {
+            if (QScreen *sc = win->screen())
+                ag = sc->availableVirtualGeometry().size();
+        }
+    }
+    resize(sizeHint().boundedTo(ag));
```

Comment:
What will happen if ag can't be retrieved? (e.g., `parent`, `win` or `sc` is `nullptr`)

Reply:
> What will happen if ag can't be retrieved?

Nothing! Because the window cannot be made smaller than its size hint.

Anyhow, `ag` can always be retrieved. Here, the above conditions make sense only logically, not practically.

Reply:
The following explanation is only for the sake of completeness:

Conditions like the above ones are necessary for windows/widgets that can appear on logging (like Panel) in or may change size on logging out (like desktop). It's fascinating how unusual the screen may be in those moments ;)

That's not the case with modal dialogs because their parents are already "formed" completely.